mlos_bench/mlos_bench/tests/services/remote/ssh/test_ssh_host_service.py

```diff
@@ -0,0 +1,207 @@
+        },
+    )
+    assert status == Status.PENDING
+    assert "asyncRemoteExecResultsFuture" in results_info
+    status, results = ssh_host_service.get_remote_exec_results(results_info)
+    assert status == Status.SUCCEEDED
+    assert results["stdout"].strip() == ALT_TEST_SERVER_NAME
+
+    # Test reusing the existing connection.
+    (status, results_info) = ssh_host_service.remote_exec(
+        script=["echo BAR=$BAR && echo UNUSED=$UNUSED && false"],
+        config=config,
+        # Also test interacting with environment_variables.
+        env_params={
+            "BAR": "bar",
+        },
+    )
+    status, results = ssh_host_service.get_remote_exec_results(results_info)
+    assert status == Status.FAILED  # should retain exit code from "false"
+    stdout = str(results["stdout"])
+    assert stdout.splitlines() == [
+        "BAR=bar",
+        "UNUSED=",
+    ]
+    connection, client = ssh_host_service._event_loop_thread_ssh_client_cache._cache[connection_id]
+    assert connection._local_port == local_port
+
+    # Close the connection (gracefully)
+    connection.close()
+
+    # Try and reconnect and see if it detects the closed connection and starts over.
+    (status, results_info) = ssh_host_service.remote_exec(
+        script=[
+            # Test multi-string scripts.
+            "echo FOO=$FOO\n",
+            # Test multi-line strings.
+            "echo BAR=$BAR\necho BAZ=$BAZ",
+        ],
+        config=config,
+        # Also test interacting with environment_variables.
+        env_params={
+            'FOO': 'foo',
+        },
+    )
+    status, results = ssh_host_service.get_remote_exec_results(results_info)
+    assert status == Status.SUCCEEDED
+    stdout = str(results["stdout"])
+    lines = stdout.splitlines()
+    assert lines == [
+        "FOO=foo",
+        "BAR=",
+        "BAZ=",
+    ]
+    # Make sure it looks like we reconnected.
+    connection, client = ssh_host_service._event_loop_thread_ssh_client_cache._cache[connection_id]
+    assert connection._local_port != local_port
+
+
+@requires_docker
+@pytest.mark.parametrize("graceful", [True, False])
+@pytest.mark.xdist_group("ssh_test_server")
+def test_ssh_service_reboot(docker_services: DockerServices,
+                            alt_test_server: SshTestServerInfo,
+                            ssh_host_service: SshHostService,
+                            graceful: bool) -> None:
+    """
+    Test the SshHostService reboot.
+    """
+    # Note: rebooting changes the port number unfortunately, but makes it
+    # easier to check for success.
+    # Also, it may cause issues with other parallel unit tests, so we run it as
+    # a part of the same unit test for now.
+    alt_test_server_ssh_service_config = alt_test_server.to_ssh_service_config()
+    (status, results_info) = ssh_host_service.remote_exec(
+        script=[
+            "echo \"sleeping...\"",
+            "sleep 30",
+            "echo \"shouldn't reach this point\""
```

Comment:
```suggestion
            'echo "sleeping..."',
            'sleep 30',
            'echo "shouldn't reach this point"'
```
but maybe `\"` is more explicit